src/adapter/breakpoints/entryBreakpoint.ts

```diff
@@ -2,15 +2,40 @@
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
-import { Breakpoint } from './breakpointBase';
-import { BreakpointManager } from '../breakpoints';
+import { Breakpoint, LineColumn } from './breakpointBase';
+import { BreakpointManager, EntryBreakpointMode } from '../breakpoints';
 import Dap from '../../dap/api';
+import { basename, extname } from 'path';
+import { Thread } from '../threads';
 
 /**
  * A breakpoint set automatically on module entry.
  */
 export class EntryBreakpoint extends Breakpoint {
-  constructor(manager: BreakpointManager, source: Dap.Source) {
+  public static getModeKeyForSource(mode: EntryBreakpointMode, path: string) {
+    return mode === EntryBreakpointMode.Greedy ? basename(path, extname(path) || undefined) : path;
+  }
+
+  constructor(
+    manager: BreakpointManager,
+    source: Dap.Source,
+    private readonly mode: EntryBreakpointMode,
+  ) {
     super(manager, source, { lineNumber: 1, columnNumber: 1 });
   }
+
+  protected _setPredicted() {
+    return Promise.resolve();
```

Comment:
Since I was here, I stubbed this out -- there's no point to looking at sourcemaps while setting entry BPs, since these BPs are exclusively used for intercepting scripts that don't yet have sourcemaps.